packages/cli/test/integration/folder/folder.controller.test.ts

```diff
@@ -1210,327 +1210,61 @@ describe('GET /projects/:projectId/folders', () => {
-		expect(response.body.data).toHaveLength(2);
-		expect(response.body.data.map((f: any) => f.name)).toEqual(['Folder 4', 'Folder 3']);
-	});
-
-	test('should sort folders by name ascending', async () => {
-		await createFolder(ownerProject, { name: 'Z Folder' });
-		await createFolder(ownerProject, { name: 'A Folder' });
-		await createFolder(ownerProject, { name: 'M Folder' });
-
-		const response = await authOwnerAgent
-			.get(`/projects/${ownerProject.id}/folders`)
-			.query({ sortBy: 'name:asc' })
-			.expect(200);
-
-		expect(response.body.data.map((f: any) => f.name)).toEqual([
-			'A Folder',
-			'M Folder',
-			'Z Folder',
-		]);
-	});
-
-	test('should sort folders by name descending', async () => {
-		await createFolder(ownerProject, { name: 'Z Folder' });
-		await createFolder(ownerProject, { name: 'A Folder' });
-		await createFolder(ownerProject, { name: 'M Folder' });
-
-		const response = await authOwnerAgent
-			.get(`/projects/${ownerProject.id}/folders`)
-			.query({ sortBy: 'name:desc' })
-			.expect(200);
-
-		expect(response.body.data.map((f: any) => f.name)).toEqual([
-			'Z Folder',
-			'M Folder',
-			'A Folder',
-		]);
-	});
-
-	test('should sort folders by updatedAt', async () => {
-		await createFolder(ownerProject, {
-			name: 'Older Folder',
-			updatedAt: DateTime.now().minus({ days: 2 }).toJSDate(),
-		});
-		await createFolder(ownerProject, {
-			name: 'Newest Folder',
-			updatedAt: DateTime.now().toJSDate(),
-		});
-		await createFolder(ownerProject, {
-			name: 'Middle Folder',
-			updatedAt: DateTime.now().minus({ days: 1 }).toJSDate(),
-		});
-
-		const response = await authOwnerAgent
-			.get(`/projects/${ownerProject.id}/folders`)
-			.query({ sortBy: 'updatedAt:desc' })
-			.expect(200);
-
-		expect(response.body.data.map((f: any) => f.name)).toEqual([
-			'Newest Folder',
-			'Middle Folder',
-			'Older Folder',
-		]);
-	});
-
-	test('should select specific fields when requested', async () => {
-		await createFolder(ownerProject, { name: 'Test Folder' });
+	test('should return folder content if user has project:viewer role in team project', async () => {
+		const project = await createTeamProject('test project', owner);
+		await linkUserToProject(member, project, 'project:viewer');
+		const folder = await createFolder(project, { name: 'Test Folder' });
 
-		const response = await authOwnerAgent
-			.get(`/projects/${ownerProject.id}/folders?select=["id","name"]`)
+		const response = await authMemberAgent
+			.get(`/projects/${project.id}/folders/${folder.id}/content`)
 			.expect(200);
 
-		expect(response.body.data[0]).toEqual({
-			id: expect.any(String),
-			name: 'Test Folder',
-		});
-
-		// Other fields should not be present
-		expect(response.body.data[0].createdAt).toBeUndefined();
-		expect(response.body.data[0].updatedAt).toBeUndefined();
-		expect(response.body.data[0].parentFolder).toBeUndefined();
+		expect(response.body.data.totalWorkflows).toBeDefined();
```

Comment:
Always checking at tests like these for the actual amounts would be better for future refactoring safety, but yeah I get that the counts are zero here.